File: BE/Seguridad/PatenteAbsBE.vb
Imports System.Windows.Forms

Public MustInherit Class PatenteAbsBE
    Inherits ID

    Public Property nombre As String


    Public Overrides Function ToString() As String
        Return Me.nombre
    End Function

    Public MustOverride Function Permite(formulario As String) As Boolean
    Public MustOverride Function ListarPatentesHojasPermitidas() As HashSet(Of PatenteBE)
End Class

File: BE_2/Seguridad/PatenteAbsBE.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using System.Windows.Forms;

public abstract class PatenteAbsBE : ID
{
    public string nombre { get; set; }


    public override string ToString()
    {
        return this.nombre;
    }

    public abstract bool Permite(string formulario);
    public abstract HashSet<PatenteBE> ListarPatentesHojasPermitidas();
}
